Creates EmploymentDetail model

mohinid opened the pull request:
## Notion card
https://www.notion.so/saeloun/Add-tables-for-personal-employment-Allocated-devices-compensation-71ef49d0c38e4242bd2998120141372d

## Summary
1. Creates employment_details table with appropriate columns.
2. Creates 1-1 association between CompanyUser and EmploymentDetail model.
3. Adds validation for the EmploymentDetail model.
4. Add RSpec for both EmploymentDetail and CompanyUser models.


## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
Added validations & RSpecs for User and EmploymentDetail models.

### Checklist:

- [x] I have manually tested all workflows
- [x] I have performed a self-review of my own code
- [x] I have added automated tests for my code

rohitjoshixyz commented on spec/models/employment_detail_spec.rb:
```diff
@@ -0,0 +1,26 @@
+      it { is_expected.to belong_to(:user) }
+    end
+
+    describe "validate presence of" do
```
```suggestion
    describe "Presence" do
```

rohitjoshixyz commented on spec/models/employment_detail_spec.rb:
```diff
@@ -0,0 +1,26 @@
+      it { is_expected.to validate_presence_of(:employee_id) }
+    end
+
+    describe "validate comparisons" do
+      it "resignation date should be later than joining date" do
+        expect(employment_detail.resigned_at).to be > employment_detail.joined_at
+      end
+    end
```
What happens when either of them is `nil`? Add a spec for each

mohinid replied:
Joined_at cannot be nill. I have validation added for the same. Will add test for nil resigned_at date.

rohitjoshixyz commented on spec/factories/employment_details.rb:
```diff
@@ -0,0 +1,12 @@
+    user
+    designation { "SDE" }
+    employment_type { "Salaried" }
+    joined_at { Faker::Date.between(from: "2020-01-01", to: "2021-01-01") }
+    resigned_at { Date.today }
```
This factory will always create a resigned employee by default.
Let's keep `resigned_at` nil by default

rohitjoshixyz reviewed (commented):
@mohinid Added a few suggestions

akhilgkrishnan commented on app/models/user.rb:
```diff
@@ -60,6 +60,7 @@ class User < ApplicationRecord
   has_many :identities, dependent: :delete_all
   has_one_attached :avatar
   rolify strict: true
+  has_one :employment_detail, dependent: :destroy
```
can you move this on line:62

mohinid replied:
done

akhilgkrishnan reviewed (commented):
@mohinid I have got a couple of doubts related to this flow because, As per the Miru workflow the User can be part of multiple companies and the employee details vary with the position handled in each company. So we should add a scope for the company model.

mohinid commented:
> @mohinid I have got a couple of doubts related to this flow because, As per the Miru workflow the User can be part of multiple companies and the employee details vary with the position handled in each company. This EmploymentDetails model hasn't given an idea about the company which the employee is associated with. So we should add a scope for the company model.

Hi @akhilgkrishnan , let's discuss today.

rohitjoshixyz commented:
@akhilgkrishnan Good catch: we could associate the `employment_details` table with the `company_users` table instead of the `user` model, which should address this concern @mohinid

mohinid commented:
Removed User reference and added reference of company_detail with employment_detail. @akhilgkrishnan  @rohitjoshixyz 
All tests passed. Please review.

<img width="910" alt="Screenshot 2022-06-13 at 3 14 13 PM" src="https://user-images.githubusercontent.com/5313625/173327139-ff762925-3425-4f95-8995-225324fa8abb.png">

rohitjoshixyz reviewed (approved):
LGTM 🎉

akhilgkrishnan reviewed (approved):
LGTM. Thanks @mohinid for making the requested changes
